tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -778,6 +778,8 @@ def shard(self, num_shards, index):
 
     - Be sure to shard before you use any randomizing operator (such as
       shuffle).
+    - Invoking repeat before shuffle may blur epoch boundaries. Consider
```

Comment:
Should this be "Invoking repeat before *shard*"?

Reply:
Technically, both are true.
* `repeat` before `shard` can bring into the current epoch elements from the next epoch.
* `repeat` before `shuffle` can move elements in and out of the current epoch.

The intention was to call out the latter (hence the suggestion of `shuffle_and_repeat`). While not necessarily relevant to `shard` itself and thus perhaps it should be omitted for brevity, the sample code may result in certain undesirable effects that the end user should be aware of. While this is explicitly addressed in the Performance Guide, I'm not sure if I've seen it anywhere else (e.g., documentation for `repeat` or `shuffle`).

Regarding the former point, it is implicitly addressed by the bullet below but I can call it out directly to make it clear if desired.

_TL;DR_: It's a minor documentation update trying to point users in the right direction, so feel free to close if you feel it's unnecessary (it very well may be) or suggest a change that might be more helpful.

Reply:
Sure, it's a fine piece of advice, but it seems out of place here in the documentation for `Dataset.shard()`.

Perhaps moving it to the `Dataset.shuffle()` docstring would make sense? Or else there might be a [spot in the programmer's guide](https://www.tensorflow.org/guide/datasets#randomly_shuffling_input_data) where the point could be driven home? (Also, rather than pointing to the `tf.contrib.data.shuffle_and_repeat()` API, which is going to be deprecated along with the rest of `tf.contrib` soon, maybe just say "apply `shuffle()` before `repeat()` if this is undesirable."